test/integration/render/lifecycle.spec.tsx

```diff
@@ -0,0 +1,182 @@
+import plusnew, { Props, component, store } from 'index';
+
+describe('lifecycle', () => {
+  let container: HTMLElement;
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    container.innerHTML = 'lots of stuff';
+    document.body.appendChild(container);
+  });
+
+  describe('componentDidMount', () => {
+    it('basic hook doing nothing', () => {
+      const componentDidMountSpy = jasmine.createSpy('componentDidMountSpy');
+
+      const NestedComponent = component(
+        'Component',
+        (_Props: Props<{}>, componentInstance) => {
+          componentInstance.registerLifecycleHook('componentDidMount', componentDidMountSpy);
+
+          return <div />;
+        },
+      );
+
+      const local = store(false);
+
+      const MainComponent = component(
+        'Component',
+        () => <local.Observer>{local => local && <NestedComponent />}</local.Observer>,
+      );
+
+      plusnew.render(<MainComponent />, container);
+
+      expect(container.childNodes.length).toBe(0);
+      expect(componentDidMountSpy.calls.count()).toBe(0);
+
+      local.dispatch(true);
+
+      expect(container.childNodes.length).toBe(1);
+      expect((container.childNodes[0] as HTMLElement).tagName).toBe('DIV');
+      expect(componentDidMountSpy.calls.count()).toBe(1);
+    });
+
+    it('hook doing sideeffect to parent', () => {
```

Comment:
more meaningful? "First nested component dispatches a value to local store in parent and gets rerendered and displayed in DOM only once."